suite/runsuite_wrapper.pl

```diff
@@ -150,11 +152,33 @@
                                    'code_api|tool.drcachesim.invariants' => 1, # i#2892
                                    'code_api|tool.drcacheoff.simple' => 1,
                                    'code_api|tool.histogram.gzip' => 1);
-            # FIXME i#2417: fix flaky AArch64 tests
-            %ignore_failures_64 = ('code_api|linux.sigsuspend' => 1,
-                                   'code_api|pthreads.pthreads_exit' => 1);
+            if ($needs_unsafe_ldstex) {
+                # FIXME i1698: ldrex..strex pair constraints
+                %ignore_failures_64 = ('code_api|linux.child-whitelist' => 1,
+                                      'code_api|linux.sigsuspend' => 1,
+                                      'code_api|linux.threadexit2' => 1,
+                                      'code_api|linux.vfork' => 1,
+                                      'code_api|pthreads.ptsig' => 1,
+                                      'code_api|pthreads.pthreads_exit' => 1,
+                                      'code_api|code_api|pthreads.pthreads' => 1,
+                                      'code_api|client.truncate' => 1,
+                                      'code_api|client.drmgr-test' => 1,
+                                      'code_api|client.drx_buf-test' => 1,
+                                      'code_api|tool.drcacheoff.simple' => 1,
+                                      'code_api|tool.drcacheoff.multiproc' => 1,
+                                      'code_api|tool.drcacheoff.opcode_mix' => 1,
+                                      'code_api|tool.drcacheoff.view' => 1,
+                                      'code_api|tool.histogram.offline' => 1);
```

Comment:
Hmm, so all these fail even with -unsafe_build_ldstex?  I could see the tracing ones failing due to the instrumentation not interacting well with the mega-block??  Though a lot of these have a low bar to passing: missing some references in the trace shouldn't matter.  And what about non-instrumentation ones like vfork or sigsuspend?  Do we know why those fail?

I don't see these listed in #1698: maybe you could add this list there with an explanation of the failures?